### PR TITLE
Api/test

### PR DESCRIPTION
**Pull request by ovekyc**
This is api test base code 
configure was changed config/test.json

please checkout those files
src/test/node/items/controllers.js
src/server/items/controllers.js

### CODE REVIEW

File: config/default.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "Example Project",
```
Comment (fguy) [inline config/default.json]: Please change this name.

File: src/server/items/controllers.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import db from '../database';
+
+
+export default {
+  getAll: (res, cb) => {
+    let result = {};
```
Comment (fguy) [inline src/server/items/controllers.js]: Please use "const" unless it's reassigned.

File: src/server/routes/api-routes.js [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import express from 'express';
-import items from '../items/items-routes';
+import items from '../items/routes';
```
Comment (fguy) [inline src/server/routes/api-routes.js:2]: 👍

File: src/test/node/items/controllers.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import test from 'tape';
+import ItemController from '../../../server/items/controllers';
+import httpMocks from 'node-mocks-http';
+import testDB from '../../../server/database';
+
+const item1 = {
```
Comment (fguy) [inline src/test/node/items/controllers.js]: Don't use meaningless number suffix. item_red_selo would be better.

File: src/test/node/items/controllers.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  description: 'This is Red Selo',
+  lat: 30.565398,
+  lng: 126.9907941,
+  address: 'Red Selo',
+  createdDate: 'Wed Mar 25 2015 09:00:00 GMT+0900 (KST)',
+  modifiedDate: 'Wed Mar 25 2015 09:00:00 GMT+0900 (KST)',
+  category: 'default'
+};
+const item2 = {
```
Comment (fguy) [inline src/test/node/items/controllers.js]: ditto

File: src/test/node/items/controllers.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  lat: 37.565398,
+  lng: 126.9907941,
+  address: 'Alaska',
+  createdDate: 'Wed Mar 25 2015 09:00:00 GMT+0900 (KST)',
+  modifiedDate: 'Wed Mar 25 2015 09:00:00 GMT+0900 (KST)',
+  category: 'default'
+};
+
+test('get all items from database', t => {
+  let ops = [
```
Comment (fguy) [inline src/test/node/items/controllers.js]: "const ops"

File: src/server/items/controllers.js (hunk 2 above)
Comment (ovekyc) [reply src/server/items/controllers.js]: Oh the value of result is changed in line 11

File: src/server/items/controllers.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import db from '../database';
+
+export default {
+  getAll: (res, cb) => {
+    let result = {};
+    db.createReadStream({
+      start: 'item1',
+      end: 'item' + '\xFF'
+    }).on('data', (data) => {
+      result[data.key] = data.value;
```
Comment (fguy) [inline src/server/items/controllers.js:10]: This is not a reassignment. The result is using initial memroy address.

Comment (ovekyc) [reply src/server/items/controllers.js:10]: Ahhh i got it!

**Comment by fguy** [comment]
LGTM

**Comment by fguy** [comment]
:+1

**Comment by fguy** [comment]
LG2M

**Comment by fguy** [comment]
LGTM

**Pull request merged**